python/cudf/cudf/core/reshape.py

```diff
@@ -13,7 +13,34 @@
 _axis_map = {0: 0, 1: 1, "index": 0, "columns": 1}
 
 
-def concat(objs, axis=0, ignore_index=False, sort=None):
+def align_objs(objs):
+    """Align a set of Series or Dataframe objects.
+
+    Parameters
+    ----------
+    objs : list of DataFrame, Series, or Index
+
+    Returns
+    -------
+    A set of reindexed and aligned objects ready for concatenation
+    """
+    index = objs[0].index
+    for obj in objs[1:]:
+        name = index.name
+        index = (
+            cudf.DataFrame(index=obj.index)
+            .join(cudf.DataFrame(index=index), how="outer")
+            .index
+        )
+        index.name = name
+
+    for i, obj in enumerate(objs):
+        objs[i] = obj.reindex(index)
```

Comment:
This assumes a list is passed as input to `align_objs` which may not be true. May want to just create a new list to return here.

Reply:
This doesn't solve the problem. If `objs` is a tuple for example it's immutable and you can't change it's contents like you're trying to do.

Reply:
Like this?

```
   a_objs = []
    for i, obj in enumerate(objs):
        a_objs.append(obj.reindex(index))

    return a_objs
```

Reply:
Yup something like that looks good